ban/commands/municipality.py

```diff
@@ -0,0 +1,86 @@
+from ban.commands import command, reporter
+from ban.core import models
+from ban.core import versioning
+from . import helpers
+
+
+@command
+@helpers.session
+def merge(destination, sources=[], name='', label='', **kwargs):
+    if destination in sources:
+        helpers.abort('Destination in sources')
+    if name == '':
+        helpers.abort('Name should not be empty')
+    if label == '':
+        helpers.abort('Label should not be empty')
+    try:
+        destination = models.Municipality.get(
+            models.Municipality.insee == destination)
+    except models.Municipality.DoesNotExist:
+        helpers.abort('Destination does not exist')
+    if not sources:
+        helpers.abort('No sources')
+    sources_inst = []
+    # Make sure all sources exist before processing any of them.
+    for source in sources:
+        try:
+            source = models.Municipality.get(
+                models.Municipality.insee == source)
+        except models.Municipality.DoesNotExist:
+            helpers.abort('Source {} does not exist'.format(source))
+        else:
+            sources_inst.append(source)
+    source_done = []
+    areas = []
+    for source in sources_inst:
+        if source.insee not in source_done:
```

Comment:
Good point, but imho you should deduplicate the sources while they still are INSEE values instead, by using a `set`.